generateBuildrootSBOM.py

```diff
@@ -44,23 +44,24 @@ def create_buildroot_sbom(args, br_bom):
     # the resulting JSON. Use a try/except block to help with run time issues.
     with open(args.input_file, newline='') as csvfile:
         sheetX = csv.DictReader(csvfile)
-        my_component_list = []
+
         for row in sheetX:
             try:
-                purl_info: str | Any = "pkg:generic/" + row['PACKAGE'] + "@" + row['VERSION'] + \
-                                       "?download_url=" + row['SOURCE SITE'] + row['SOURCE ARCHIVE']
+                # purl_info: str | Any = "pkg:generic/" + row['PACKAGE'] + "@" + row['VERSION'] + \
+                #                       "?download_url=" + row['SOURCE SITE'] + row['SOURCE ARCHIVE']
 
                 from packageurl import PackageURL
 
+                purl_info = PackageURL(type='generic', name=row['PACKAGE'], version=row['VERSION'],
+                                       qualifiers={'download_url': row['SOURCE SITE'] + row['SOURCE ARCHIVE']})
+
                 componenttype = cyclonedx.model.component.ComponentType('firmware')
+                this_component_license = [LicenseChoice(expression=(row['LICENSE']))]
```

Comment:
:x: this is wrong.
from your example files [you posted on slack](https://cyclonedx.slack.com/archives/C053DUNL7GE/p1686081821943049) i see the values of `row['LICENSE']` would be 
- `unknown`
- `BSL-1.0`
- `Python-2.0, others	`
- `Zlib`
- `GPL-2.0 or BSD-3-Clause	`

the above values - some of them are valid SPDX license expressions, some are SPDX license ids, some are none of them.

you are coding it as if all were license expressions. this is just not true.

fix this as described here: https://github.com/CycloneDX/cyclonedx-buildroot/pull/9#discussion_r1221103461 and https://github.com/CycloneDX/cyclonedx-buildroot/pull/9#discussion_r1221613248

Reply:
fixed via 8109b7d6e2150446673802afb4e2922d086288be